tests/StackExchange.Redis.Tests/Strings.cs

```diff
@@ -68,6 +68,76 @@ public async Task Set()
             }
         }
 
+        [Fact]
+        public async Task GetEx_NoValue()
+        {
+            using var muxer = Create();
+            Skip.IfMissingFeature(muxer, nameof(RedisFeatures.GetEx), r => r.GetEx);
+
+            var conn = muxer.GetDatabase();
+            var key = Me();
+            conn.KeyDelete(key, CommandFlags.FireAndForget);
+
+            var empty_val = await conn.StringGetSetExpiryAsync(key, TimeSpan.FromHours(1));
+
+            Assert.False(empty_val.HasValue);
+        }
+
+        [Fact]
+        public async Task GetEx_Relative()
+        {
+            using var muxer = Create();
+            Skip.IfMissingFeature(muxer, nameof(RedisFeatures.GetEx), r => r.GetEx);
+
+            var conn = muxer.GetDatabase();
+            var key = Me();
+            conn.KeyDelete(key, CommandFlags.FireAndForget);
+
+            conn.StringSet(key, "abc", TimeSpan.FromHours(1));
+            var relative_sec = conn.StringGetSetExpiryAsync(key, TimeSpan.FromMinutes(30));
+            var relative_sec_ttl = conn.KeyTimeToLiveAsync(key);
+
+            Assert.Equal("abc", await relative_sec);
+            var time = await relative_sec_ttl;
+            Assert.True(time > TimeSpan.FromMinutes(29.9) && time <= TimeSpan.FromMinutes(30));
+        }
+
+        [Fact]
+        public async Task GetEx_Absolute()
+        {
+            using var muxer = Create();
+            Skip.IfMissingFeature(muxer, nameof(RedisFeatures.GetEx), r => r.GetEx);
+            var conn = muxer.GetDatabase();
+            var key = Me();
+            conn.KeyDelete(key, CommandFlags.FireAndForget);
+
+            conn.StringSet(key, "abc", TimeSpan.FromHours(1));
+            var val = conn.StringGetSetExpiryAsync(key, DateTime.UtcNow.AddMinutes(30));
+            var val_ttl = conn.KeyTimeToLiveAsync(key);
+
+            Assert.Equal("abc", await val);
+            var time = await val_ttl;
+            Assert.True(time > TimeSpan.FromMinutes(29.9) && time <= TimeSpan.FromMinutes(30));
+        }
+
+        [Fact]
+        public async Task GetEx_Persist()
+        {
+            using var muxer = Create();
+            Skip.IfMissingFeature(muxer, nameof(RedisFeatures.GetEx), r => r.GetEx);
+
+            var conn = muxer.GetDatabase();
+            var key = Me();
+            conn.KeyDelete(key, CommandFlags.FireAndForget);
+
+            conn.StringSet(key, "abc", TimeSpan.FromHours(1));
+            var val = conn.StringGetSetExpiryAsync(key, null);
+            var val_ttl = conn.KeyTimeToLiveAsync(key);
+
+            Assert.Equal("abc", await val);
+            Assert.Null(await val_ttl);
+        }
+
```

Comment:
I'd Expect one more test here to try out the path in which you'd throw an [ArgumentException](https://github.com/StackExchange/StackExchange.Redis/pull/1743/files#diff-4dd7cb0578bed18d8797224b2f3a47777d9592d24137357eb7eac71d162508dbR3564) in the case of an `TimeStampKind.Unspecified` timestamp. TBH I'm not even sure what that means :)